bcp47/tests/Data/BCP47/TrieSpec.hs

```diff
@@ -6,14 +8,17 @@ import Prelude hiding (lookup)
 
 import Data.BCP47
 import Data.BCP47.Trie
+import qualified Data.List.NonEmpty as NE
 import Test.Hspec
 import Test.QuickCheck
+import Test.QuickCheck.Modifiers (NonEmptyList(NonEmpty))
 
 spec :: Spec
 spec = do
   describe "Trie" $ do
-    it "has equality" $ property $ \xs ->
-      fromList xs `shouldBe` (fromList xs :: Trie Bool)
+    it "has equality" $ property $ \(NonEmpty xs) ->
+      fromList (NE.fromList xs)
+        `shouldBe` (fromList (NE.fromList xs) :: Trie Bool)
```

Comment:
The original code here seems like it should've worked. Since you added the `NonEmpty xs` pattern and then offsetting `NE.fromList xs`-s at every use. I'm curious if you did this because there's an error (or ambiguity) I'm missing, or just for the benefit of the reader to bring attention to it. If it's the latter (and maybe even the former) I think `TypeApplications` might be nice.

Reply:
`NonEmpty` doesn't have an `Arbitrary` instance.

Reply:
What is `property $ \(NonEmpty xs)` -> doing?

Reply:
Oh god, it's a different `NonEmpty`. Well that's confusing -- anyway, carry on.